### DISCUSSION TITLE
Deploying to AWS Lambdas (mkdir '/var/task/temp' error)

### DISCUSSION THREAD

**Discussion by gusgorman2** [mikro-orm/mikro-orm#5042] [Q&A]
Hello,

Is anyone deploying an app with Mikro ORM to lambdas?

The only thread I can find that mentions lambdas is this [one](https://github.com/mikro-orm/mikro-orm/discussions/2613) 

At the moment we have this error: `Error: ENOENT: no such file or directory, mkdir '/var/task/temp' `which we haven't investigated yet but I'm guessing is Mikro ORM creating the cache, which perhaps we could get round by deploying it pre-built as documented here: https://mikro-orm.io/docs/deployment ?

Anyone have any experience of this?

**Comment by B4nan** [comment]
The `temp` folder sounds like metadata cache, I guess you use ts-morph for metadata reflection? As that is the only one that uses caching.

I'd suggest you try v6 if you want to use ts-morph, as it allows to compile the metadata cache into a single file which you can import in your app to remove dynamic imports. It should be much more suitable for compilation and lambdas in general due to this (as well as [other improvements](https://mikro-orm.io/docs/next/upgrading-v5-to-v6#removal-of-static-require-calls)).

https://mikro-orm.io/docs/next/deployment#deploy-pre-built-cache

(I've never deployed anything to lambda or similar myself, so can't help much with this topic in general.)

**Reply by gusgorman2** [reply]
Great thanks, and yes we are using TsMorph
